dot/sync/chain_sync.go

```diff
@@ -932,6 +936,34 @@ func (cs *chainSync) validateJustification(bd *types.BlockData) error {
 	return nil
 }
 
+func (cs *chainSync) getHighestBlock() (int64, error) {
+	cs.RLock()
+	ps := cs.peerState
+	cs.RUnlock()
+
+	if len(ps) == 0 {
+		return 0, errNoPeers
+	}
+
+	highestBlock := big.NewInt(-1)
+
+	for _, ps := range ps {
+		if ps.number == nil {
+			continue
+		}
+		if ps.number.Cmp(highestBlock) < 0 { // ps.number < highestBlock: https://pkg.go.dev/math/big#Int.Cmp
+			continue
+		}
+		highestBlock = ps.number
+	}
+
+	if highestBlock.Cmp(big.NewInt(-1)) == 0 { // highestBlock == -1 : https://pkg.go.dev/math/big#Int.Cmp
+		return 0, errNilBlockData
```

Comment:
you can probably just return 0 in this case, or check it in the higher up RPC function and return 0 there.

Reply:
ah I see you return 0 in the higher function, so that's fine

Reply:
Could we resolve this conversation? 😛

Reply:
Done in 60eaaa8